refinery/tool_manager/models.py

```diff
@@ -305,6 +307,19 @@ def update_file_relationships_string(self):
         self.save()
 
 
+@receiver(post_delete, sender=Tool)
```

Comment:
If container is created after tool instance, I'd make it a pre_delete handler, so container would be stopped before tool instance is deleted to preserve logical order.